code/lib/builder-webpack5/src/index.ts

```diff
@@ -219,10 +218,9 @@ const builder: BuilderFunction = async function* builderGeneratorFn({ startTime,
 
       logger.trace({ message: '=> Preview built', time: process.hrtime(startTime) });
       if (stats && stats.hasWarnings()) {
-        // @ts-ignore (FIXME should be @ts-expect-error but fails build --prep)
         stats
           .toJson({ warnings: true } as StatsOptions)
-          .warnings.forEach((e) => logger.warn(e.message));
+          .warnings?.forEach((e) => logger.warn(e.message));
```

Comment:
TypeScript doesn't understand the relation between `hasWarnings` and `warnings` being defined. a non-null assertion could alternatively be used here.

Reply:
or just an additional check on L220 !stats.warnings